frontend/src/metabase/admin/settings/notifications/WebhookForm.tsx

```diff
@@ -0,0 +1,212 @@
+import type { FormikHelpers } from "formik";
+import { t } from "ttag";
+import * as Yup from "yup";
+
+import { useTestChannelMutation } from "metabase/api/channel";
+import {
+  Form,
+  FormChipGroup,
+  FormProvider,
+  FormSubmitButton,
+  FormTextInput,
+} from "metabase/forms";
+import { FormKeyValueMapping } from "metabase/forms/components/FormKeyValueMapping";
+import { useActionButtonLabel } from "metabase/hooks/use-action-button-label";
+import { Button, Chip, Flex, Alert, Text, Group, Icon } from "metabase/ui";
+import type { NotificationAuthMethods } from "metabase-types/api";
+
+const validationSchema = Yup.object({
+  url: Yup.string()
+    .url(t`Please enter a correctly formatted URL`)
+    .required(),
+  name: Yup.string().required(t`Please add a name`),
+  description: Yup.string().required(t`Please add a description`),
+  "auth-method": Yup.string()
+    .required()
+    .equals(["none", "header", "query-param", "request-body"]),
+  "auth-info": Yup.object(),
+});
+
+const styles = {
+  input: {
+    fontWeight: 400,
+  },
+  label: {
+    fontSize: "14px",
+    marginBottom: "0.75rem",
+  },
+};
+
+export type WebhookFormProps = {
+  url: string;
+  name: string;
+  description: string;
+  "auth-method": NotificationAuthMethods;
+  "auth-info": Record<string, string>;
+};
+
+type WebhookFormikHelpers = FormikHelpers<WebhookFormProps>;
+
+export const handleFieldError = (e: any) => {
+  if (!e.data) {
+    return;
+  } else if (typeof e.data === "string") {
+    throw { data: { errors: { url: e.data } } };
+  } else if (e.data.message) {
+    throw { data: { errors: { url: e.data.message } } };
+  } else if (typeof e.data.errors === "object") {
+    throw e;
+  }
+};
+
+export const WebhookForm = ({
+  onSubmit,
+  onCancel,
+  onDelete,
+  initialValues,
+  submitLabel = t`Create destination`,
+}: {
+  onSubmit: (props: WebhookFormProps) => void;
+  onCancel: () => void;
+  onDelete?: () => void;
+  initialValues: WebhookFormProps;
+  submitLabel?: string;
+}) => {
+  const { label: testButtonLabel, setLabel: setTestButtonLabel } =
+    useActionButtonLabel({ defaultLabel: t`Send a test` });
+  const [testChannel] = useTestChannelMutation();
+
+  const handleTest = async (
+    values: WebhookFormProps,
+    setFieldError: WebhookFormikHelpers["setFieldError"],
+  ) => {
+    await testChannel({
+      details: {
+        url: values.url,
+        "auth-method": values["auth-method"],
+        "auth-info": values["auth-info"],
+      },
+    })
+      .unwrap()
+      .then(
+        () => {
+          setFieldError("url", undefined);
+          setTestButtonLabel(t`Success`);
+        },
+        e => {
+          setTestButtonLabel(t`Test failed`);
+          if (typeof e === "string") {
+            setFieldError("url", e);
+          } else if (typeof e.data === "string") {
+            setFieldError("url", e.data);
+          } else if (e.data?.message) {
+            setFieldError("url", e.data.message);
+          }
```

Comment:
😬  the api sends back all these error types? could they maybe narrow that for us?

Reply:
If you look at the `getResponseErrorMessage` function, I think that indicates this is a common pattern, and one that the FE can cope with, but yeah, it's a bit odd that there is so much variability in the format of an error response

Reply:
In this particular case, the API just forwards along the response that the target API sends back. I found quite a bit of variation between errors, but I'm going to check out the method that Raffi referenced above and see if I can use that instead. I agree that this is messy